bundler/helpers/v2/lib/functions/version_resolver.rb

```diff
@@ -28,11 +33,6 @@ def version_details
       # subdependency that was removed when attempting to update it.
       return nil if dep.nil?
 
-      # If the dependency is Bundler itself then we can't trust the
-      # version that has been returned (it's the version Dependabot is
-      # running on, rather than the true latest resolvable version).
-      return nil if dep.name == "bundler"
-
```

Comment:
I explain things in the commit message, but there's a small inconsistency in Bundler 2.3.20 created by https://github.com/rubygems/rubygems/pull/5784/commits/2777e79b8efe1bf77589496632c74464022f04f3 which makes the previous early `return "latest"` trigger before this one, which is the expected one. I'm not a fan of that inconsistency, but regardless of what upstream does, we can special case Bundler earlier to get the expected outcome.